ocaml/xapi/xapi.ml

```diff
@@ -941,22 +941,34 @@ let server_init() =
                                    (fun () -> Debug_populate.do_populate ~vms:1000 ~vdis_per_vm:3 ~networks:10 ~srs:10 ~tasks:1000) ]
         );
 
+        let maybe_resync_cluster_host () =
+          let host = Helpers.get_localhost ~__context in
+          begin match Xapi_clustering.find_cluster_host ~__context ~host with
+          | Some self -> ignore
+            Xapi_mgmt_iface.(wait_for_clustering_ip ~__context ~self);
+            Xapi_cluster_host.resync_host ~__context ~host
```

Comment:
Plugging the PBD already calls resync_host, so the only thing we need here is to wait for an IP.
Rename this to `maybe_wait_for_clustering_ip`.